packages/backend/src/plugins/search.ts

```diff
@@ -26,9 +27,16 @@ import { DefaultTechDocsCollator } from '@backstage/plugin-techdocs-backend';
 export default async function createPlugin({
   logger,
   discovery,
+  config,
 }: PluginEnvironment) {
   // Initialize a connection to a search engine.
-  const searchEngine = new LunrSearchEngine({ logger });
+  const searchEngine =
```

Comment:
We'd likely want to hide this somewhere in the search-node package instead.

Reply:
What degree of "hiding" do you have in mind?

From a composability perspective I would be happy if I can freely choose any engine by explicitly adding the dependency and then select it as a search engine. But it could be helpful to provide some helpers such as:
```ts
const searchEngine = ElasticSearchSearchEngine.isActive({config}) // or isConfigured or something different
    ? await ElasticSearchSearchEngine.initialize(...)
    : new LunrSearchEngine({ logger }); // just a fallback for `yarn dev` where I might not have ES configured.
```

Reply:
I don't think this comment has as much relevance any more if we are deciding to keep the implementations open and leave the configuration on the code level only. I'll only modify this example backend to check if the config is present and pick the engine based on that: 

```
  const searchEngine = config.has('search.elasticSearch')
    ? await ElasticSearchSearchEngine.initialize({
        logger,
        config,
      })
    : new LunrSearchEngine({ logger });
```

Reply:
Those helpers would basically only be useful in the example backend, and the default scaffolded one right? With the explicit intent to make it fewer steps to enable it (essentially to simplify the installation instructions) and specifically for some "golden" engines.

Otherwise it could be argued that it's probably fine to change the engine line in the backend init code when you switch over to elasticsearch.

I have this constant battle in my head, around how much to make "just config driven" and how much to just make wired up in code explicitly.